flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/FlinkStreamRuleSets.scala

```diff
@@ -211,6 +211,29 @@ object FlinkStreamRuleSets {
     ProjectWindowTableFunctionTransposeRule.INSTANCE
   )
 
+  /** RuleSet about project, but with PROJECT_FILTER_TRANSPOSE_WHOLE_EXPRESSIONS. */
+  val PROJECT_RULES_V2: RuleSet = RuleSets.ofList(
```

Comment:
PROJECT_FILTER_TRANSPOSE_WHOLE_EXPRESSIONS means push filter fields into projection. We introduce PROJECT_RULES_V2 because we meet cyclic rule problem when we change PROJECT_FILTER_TRANSPOSE to PROJECT_FILTER_TRANSPOSE_WHOLE_EXPRESSIONS directly in PROJECT_RULES. We finally fix this issue by using PROJECT_FILTER_TRANSPOSE_WHOLE_EXPRESSIONS in project_rewrite and PROJECT_FILTER_TRANSPOSE in logical rules.